src/zilliqa.c

```diff
@@ -21,6 +21,25 @@ uint8_t * getKeySeed(uint32_t index) {
     return keySeed;
 }
 
+void compressPubKey(cx_ecfp_public_key_t *publicKey) {
+    // Uncompressed key has 0x04 + X (32 bytes) + Y (32 bytes).
+    if (publicKey->W_len != 65 || publicKey->W[0] != 0x04) {
+        PRINTF("compressPubKey: Input public key is incorrect\n");
```

Comment:
Should throw instead of just keep going without bad input

Reply:
I'll fix this.